### PR TITLE
Improve telemetry details in accountant (WIP)

### PR DESCRIPTION
**Pull request by mavam**
This PR fixes the empty accounting.log and adds extra telemetry to better understand VAST's internals.

- [x] Fix accountant
- [x] Add telemetry from SOURCEs
- [ ] Add telemetry from SINKs
- [ ] Add telemetry from INDEX
- [ ] Add telemetry from ARCHIVE
- [ ] Add telemetry from EXPORTERs

### CODE REVIEW

File: libvast/vast/system/source.hpp [hunk 1]
```diff
@@ -69,14 +69,17 @@ template <class Reader>
...
   using factory_type = table_slice_builder_ptr (*)(record_type);
 
   using downstream_manager
     = caf::broadcast_downstream_manager<table_slice_ptr>;
 
   // -- constructors, destructors, and assignment operators --------------------
 
-  source_state(caf::scheduled_actor* selfptr) : self(selfptr) {
+  source_state(caf::scheduled_actor* selfptr)
+    : self{caf::actor_cast<source_actor*>(selfptr)} {
```
Comment (Neverlord) [inline libvast/vast/system/source.hpp]: Is this actually compiling? `actor_cast` is meant for converting between handle types and not for downcasting pointers. Did you try simply taking `source_actor*` as constructor argument? If that doesn't work then please use `static_cast` for downcasting. Or simply revert this set of changes (see previous comment).

File: libvast/vast/system/source.hpp [hunk 2]
```diff
@@ -69,14 +69,17 @@ template <class Reader>
 struct source_state {
   // -- member types -----------------------------------------------------------
 
+  using source_actor = caf::stateful_actor<source_state>;
```
Comment (Neverlord) [inline libvast/vast/system/source.hpp]: What's the point of this type alias? Aside from the redundant name `source_state::source_actor`, it doesn't add anything to the state class we can't do already. Seems like you're only using it for `self`, but I see no upside over just using `event_based_actor`.

**Comment by Neverlord** [comment]
@mavam I hope you don't mind that I took this over. I also had to rebase to get it compiling again.

@tobim this is good to go from my point of view. Mind giving it a review?

**Comment by mavam** [comment]
> @mavam I hope you don't mind that I took this over. I also had to rebase to get it compiling again.

Not at all, thanks for doing so! 👍

**Comment by mavam** [comment]
By the way, this PR is not yet complete: I'd like to add details from EXPORTER, INDEX, and ARCHIVE before merging.

**Comment by tobim** [comment]
I'm in favor of splitting the open tasks into new work items, so this can be merged and won't get even more stale. Then we also can extend the statistics for each component individually, as I don't see any dependencies between those.

**Comment by Neverlord** [comment]
I fully agree. This already adds value as standalone addition and we can expand on it with independent PRs to add telemetry to other components. 👍

**Pull request merged**